common/frame_options.go

```diff
@@ -717,3 +717,36 @@ func (o *FrameWaitForSelectorOptions) Parse(ctx context.Context, opts goja.Value
 
 	return nil
 }
+
+// FrameDispatchEventOptions are options for Frame.dispatchEvent.
+type FrameDispatchEventOptions struct {
+	ElementHandleDispatchEventOptions
+	Strict bool `json:"strict"`
+}
+
+// NewFrameDispatchEventOptions returns a new FrameDispatchEventOptions.
+func NewFrameDispatchEventOptions(defaultTimeout time.Duration) *FrameDispatchEventOptions {
+	return &FrameDispatchEventOptions{
+		ElementHandleDispatchEventOptions: *NewElementHandleDispatchEventOptions(defaultTimeout),
+		Strict:                            false,
+	}
+}
+
+// Parse the FrameDispatchEventOptions options from a goja value.
+func (o *FrameDispatchEventOptions) Parse(ctx context.Context, opts goja.Value) error {
+	if err := o.ElementHandleDispatchEventOptions.Parse(ctx, opts); err != nil {
+		return err
+	}
+	if !isGojaValueExists(opts) {
+		return nil
+	}
+	gopts := opts.ToObject(k6ext.Runtime(ctx))
+	for _, k := range gopts.Keys() {
+		if k == "strict" {
+			o.Strict = gopts.Get(k).ToBoolean()
+			break
```

Comment:
AFAICS we only need the `timeout` and `strict` options, for which we have [`FrameBaseOptions`](https://github.com/grafana/xk6-browser/blob/f97ca063a8a7f483faed0d2d01251d9d370c9af8/common/frame_options.go#L34-L37), where [we already parse both of them](https://github.com/grafana/xk6-browser/blob/f97ca063a8a7f483faed0d2d01251d9d370c9af8/common/frame_options.go#L174-L179). So if you embed `FrameBaseOptions` instead of `ElementHandleDispatchEventOptions`, we should be able to just return the error from `o.FrameBaseOptions.Parse(ctx, opts)` here (after proper validation as mentioned above).

Embedding `ElementHandleDispatchEventOptions` also allows passing all the `ElementHandleBasePointerOptions` like `position` and `trial`, which I think are irrelevant for `Frame.dispatchEvent`.

Reply:
The problem here is that `FrameBaseOptions` don't have `Force` and `NoWaitAfter` options (unlike the currently embedded `ElementHandleDispatchEventOptions`). So it won't work if I simply change `ElementHandleDispatchEventOptions` to `FrameBaseOptions`.

`position` and `trial` can still be valuable here to change the behavior of the `dispatchEvent` method as what the `Frame` only does is to route the call to `ElementHandle`.

How should I proceed next?

Reply:
But why do we need `Force` and `NoWaitAfter`? Those aren't options for `Frame.dispatchEvent` [according to Playwright](https://playwright.dev/docs/api/class-frame#frame-dispatch-event). Neither are `position` and `trial`.

My main issue is with repeating the parsing of `strict`, which, given that it applies to all(?) `Frame` methods, doesn't belong here.

And if we want to differ from Playwright's API, that's fine I suppose, but it adds more things to document, so I wonder if we need to.

Reply:
We need `Force` and `NoWaitAfter` in `Frame.newAction`. I've provided the options' default values and called the method. Doing so allows us only to embed `FrameBaseOptions`.

Please take a look at [the commit]([7b4df2a](https://github.com/grafana/xk6-browser/pull/405/commits/7b4df2a5f5037da827af0284366bda7b91d6a6b9)) and let me know how I proceed next.